doc/interface.md

```diff
@@ -0,0 +1,66 @@
+
+The ringbuffer crate provides four traits. In this document their methods are described.
+
+# DEFINITIONS
+
+In this document, `front` refers to the item last pushed to the queue. `back` refers to the 
+item pushed longest ago and thus the item that will be dequeud next. 
+
+Relative indexing (like with get and get_mut) works from the `back` and counts towards the front.
+`get(0)` == `back` and `get(1)` will be dequeued right after `get(0)` 
+
+# RingBuffer
+
+This trait must be implemented for any ringbuffer. It's the "base" trait.
+With it you can't actually do anything really useful, but it provides some 
+basic methods.
+
+| function | description |
+| --- | ---| 
+| `len()` | number of items in the ringbuffer |
+| `is_empty()` | returns true if the length is zero |
+| `is_full()` | returns true if the length equals the capacity |
+| `clear` | sets the length back to zero and removes all items from the ringbuffer |
+| `capacity` | the maximum number of items in the ringbuffer |
+    
+
+# ReadableRingbuffer
+
+With this trait it's possible to use the ringbuffer as the read end of a queue.
+
+| function | description |
+| --- | ---| 
+| `skip()` | Like dequeue but drops the items it removes from the ringbuffer |
+| `dequeue()` | Removes the item pushed longest ago (fifo) from the ringbuffer and returns it. |
```

Comment:
This is now called pop right? We should update this document to reflect changes made